main.js

```diff
@@ -236,30 +240,34 @@ class openknx extends utils.Adapter {
             //before composite check, date is also composite
             knxVal = new Date(state.val);
         } else if (this.gaList.getDataById(id).native.valuetype == "composite") {
-            try {
-                knxVal = JSON.parse(state.val);
-            } catch (e) {
-                this.log.warn("stateChange: unsupported value format " + state.val + " for " + ga);
-                return;
-            }
+            //input from IOB is either object or string in object notation, type of this conversion is object
+            if (typeof state.val == 'object') {
+                knxVal = state.val;
+            } else
+                try {
```

Comment:
An improvement would be to check depending on the DPT if the object has the correct  properties

Reply:
the underlying library also does some checks